examples/mayachain-amm/utils.ts

```diff
@@ -39,3 +41,23 @@ export const printQuoteSwap = (quoteSwap: QuoteSwap) => {
     warning: quoteSwap.warning,
   })
 }
+
+/**
+ * Check if a chain is EVM and supported by the protocol
+ * @param {Chain} chain to check
+ * @returns true if chain is EVM, otherwise, false
+ */
+export const isProtocolEVMChain = (chain: Chain): boolean => {
+  return [AssetETH.chain, AssetAETH.chain].includes(chain)
+}
+
+/**
+ * Check if asset is ERC20
+ * @param {Asset} asset to check
+ * @returns true if asset is ERC20, otherwise, false
+ */
+export const isProtocolERC20Asset = (asset: CompatibleAsset): asset is TokenAsset => {
```

Comment:
this function can be imported from the package 
`isProtocolERC20Asset` no need to rewrite it here

Reply:
okay let me fix this

Reply:
Btw the swap flow is meant to be like this 
Estimate swap => (in here it will check for approval via validateSwap() ) => asset needs approval ? call approve() : doSwap()

its all clearly laid out in the read me